### PR TITLE
Introduce pluggable runtime provider

### PR DESCRIPTION
**Pull request by zuston**

### CODE REVIEW

File: tony-core/src/main/java/com/linkedin/tony/FrameworkRuntimeProvider.java [hunk 1]
```diff
@@ -15,50 +15,54 @@
...
-import com.linkedin.tony.runtime.TFRuntime;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 
 public class FrameworkRuntimeProvider {
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/FrameworkRuntimeProvider.java:26]: @zuston it'd be good if you can write a dev guide (a readme file) if others want to support another framework in TonY.

**Review by oliverhu — COMMENTED** [review]
looks like a mini dependency injection..

File: tony-core/src/main/java/com/linkedin/tony/FrameworkRuntimeProvider.java (hunk 1 above)
Comment (zuston) [reply tony-core/src/main/java/com/linkedin/tony/FrameworkRuntimeProvider.java:26]: Good idea

**Comment by zuston** [comment]
> looks like a mini dependency injection..

Just refer to the implementation of the Hadoop file system that using the Java SPI.

File: tony-core/src/main/java/com/linkedin/tony/FrameworkRuntimeProvider.java (hunk 1 above)
Comment (zuston) [reply tony-core/src/main/java/com/linkedin/tony/FrameworkRuntimeProvider.java:26]: Developing doc will be attached later on another PR.

**Review by oliverhu — APPROVED** [review]

**Pull request merged**